File: check-pdf-path.js
require('dotenv').config({ path: '.env.local' });
const { MongoClient } = require('mongodb');

async function checkPdfPath() {
  const uri = process.env.MONGODB_URI;
  console.log('Connecting to:', uri?.split('@')[1] || 'MongoDB');
  
  const client = new MongoClient(uri);
  
  try {
    await client.connect();
    const db = client.db('test');
    
    // List all collections
    const collections = await db.listCollections().toArray();
    console.log('\n📚 Collections in database:');
    collections.forEach(col => console.log(`  - ${col.name}`));
    
    // Check each collection for data
    console.log('\n📊 Collection counts:');
    for (const col of collections) {
      const count = await db.collection(col.name).countDocuments();
      console.log(`  ${col.name}: ${count} documents`);
    }
    
    // Try to find interview data in any collection with 'interview' in the name
    const interviewCollections = collections.filter(c => c.name.toLowerCase().includes('interview'));
    
    if (interviewCollections.length > 0) {
      console.log('\n🔍 Interview-related collections:');
      for (const col of interviewCollections) {
        const sample = await db.collection(col.name).findOne({});
        console.log(`\n${col.name}:`);
        console.log(JSON.stringify(sample, null, 2).substring(0, 500));
      }
    }
  } finally {
    await client.close();
  }
}

checkPdfPath().catch(console.error);

File: check_pdf_path.py
import os
from pymongo import MongoClient
from dotenv import load_dotenv

# Load environment
load_dotenv('.env.local')

# Connect to MongoDB
uri = os.getenv('MONGODB_URI')
client = MongoClient(uri)
db = client['test']

# Find the interview session with PDF
session = db.interviewSessions.find_one(
    {'candidate.name': 'CYNTHIA DWAYNE'},
    {'pdfPath': 1, 'overallScore': 1, 'candidate.name': 1}
)

if session:
    print(f"Found session for: {session.get('candidate', {}).get('name')}")
    print(f"PDF Path in DB: '{session.get('pdfPath')}'")
    print(f"Overall Score: {session.get('overallScore')}")
    
    # Check if file exists at that path
    pdf_path = session.get('pdfPath')
    if pdf_path:
        # Check as stored
        full_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), pdf_path)
        print(f"\nFull path constructed: {full_path}")
        print(f"File exists: {os.path.exists(full_path)}")
        
        # Check in public directly
        public_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'public', os.path.basename(pdf_path))
        print(f"\nDirect public path: {public_path}")
        print(f"File exists: {os.path.exists(public_path)}")
else:
    print("No session found for CYNTHIA DWAYNE")
